cdci_data_analysis/analysis/exceptions.py

```diff
@@ -33,10 +35,14 @@ class RequestNotUnderstood(BadRequest):
     it is unclear for us and we want user to please clarify!
     """
 
+
+class RequestNotAuthorized(BadRequest):
```

Comment:
this could as well return 403 status? more conventional

Reply:
Yes, makes sense.

Reply:
I am struggling to understand where to set the final staus_code (403 in this case) for the response that should be returned. Is there somewhere a similar case? I noticed that every response returned when building this returns 200

Reply:
A change in there https://github.com/oda-hub/dispatcher-app/blob/e2fa1fe1622bbd9171e23a1cf48f4b084a7bb6f5/cdci_data_analysis/flask_app/dispatcher_query.py#L566 might be needed.

alternatively, we can just let it go there https://github.com/oda-hub/dispatcher-app/blob/e2fa1fe1622bbd9171e23a1cf48f4b084a7bb6f5/cdci_data_analysis/flask_app/app.py#L158

which already handles status codes for exceptions

for auth, second option can do.

but first option may be improved to handle status in any case.

two cases relevant discussed in there:

https://github.com/oda-hub/dispatcher-app/issues/25

Reply:
The first option is probably a better ( I was already trying that way), have to figure our better what I am missing.

As for the second, in this case, that errorhandler is not executed, since we directly return the Response object.